### PR TITLE
OCT-715: Replace BigNumber with BigInt

### PR DESCRIPTION
**Pull request by jmikolajczyk**

### CODE REVIEW

**Review by aziolek — CHANGES_REQUESTED** [review]
Please add signatures to the commits.

**Comment by jmikolajczyk** [comment]
> Please add signatures to the commits.

@aziolek done

**Comment by aziolek** [comment]
> > Please add signatures to the commits.
> 
> @aziolek done

All commits need to be signed, so you need for `push --force` and override he history.

File: client/src/views/AllocationView/AllocationView.tsx [hunk 1]
```diff
@@ -171,8 +171,7 @@ const AllocationView = (): ReactElement => {
       isFetchingUserAllocation ||
       !isRewardsForProposalsSet ||
       currentEpoch === undefined ||
-      (isConnected && !userAllocations && isDecisionWindowOpen && currentEpoch > 1) ||
-      !rewardsForProposalsNew
```
Comment (aziolek) [inline client/src/views/AllocationView/AllocationView.tsx:175]: That looks suspicious. I will dig deeper why it was here, maybe truly without reason.

File: client/src/components/Allocation/AllocationItemRewards/AllocationItemRewards.tsx [hunk 2]
```diff
@@ -91,21 +92,18 @@ const AllocationItemRewards: FC<AllocationItemRewardsProps> = ({
...
-          .sub(
-            proposalMatchedProposalRewards
-              ? proposalMatchedProposalRewards.matched
-              : BigNumber.from(0),
-          )
-          .abs(),
+        simulatedMatchedBigInt -
+          bigintAbs(
+            proposalMatchedProposalRewards ? proposalMatchedProposalRewards.matched : BigInt(0),
+          ),
```
[the remark this answers is not in the document]
Comment (jmikolajczyk) [reply client/src/components/Allocation/AllocationItemRewards/AllocationItemRewards.tsx:104]: Implementation mistake -> `bigintAbs` should get subtract operation as a param.

File: client/src/hooks/mutations/useLock.ts [hunk 3]
```diff
@@ -1,19 +1,18 @@
...
-): UseMutationResult<{ hash: Hash; value: BigNumber }, unknown, BigNumber> {
+  options?: UseMutationOptions<{ hash: Hash; value: bigint }, unknown, bigint>,
+): UseMutationResult<{ hash: Hash; value: bigint }, unknown, bigint> {
   const { data: walletClient } = useWalletClient();
 
   return useMutation({
     mutationFn: async value =>
       writeContractDeposits({
-        args: [BigInt(value.toHexString())],
+        args: [`0x${value.toString(16)}`],
```
[the remark this answers is not in the document]
Comment (jmikolajczyk) [reply client/src/hooks/mutations/useLock.ts:15]: nope :(

**Review by aziolek — CHANGES_REQUESTED** [review]
When AW is open but user does not have `individualReward`, in `AllocationView`:
1. Inputs are not disabled.
2. Weird `0 ETH /` is shown in place of rewards.

Image below:

image

Both problems do not happen on the current `master`:

image

File: client/src/components/Earn/EarnTipTiles/EarnTipTiles.tsx [hunk 4]
```diff
@@ -66,7 +66,7 @@ const EarnTipTiles = (): ReactElement => {
   const isAllocateRewardsTipVisible =
     (!wasAllocateRewardsAlreadyClosed &&
       isDecisionWindowOpen &&
-      individualReward !== 0n &&
+      !individualReward &&
```
Comment (aziolek) [inline client/src/components/Earn/EarnTipTiles/EarnTipTiles.tsx:69]: `isAllocateRewardsTipVisible` when the user doesn't have `individualReward`? Why?

Comment (aziolek) [reply client/src/components/Earn/EarnTipTiles/EarnTipTiles.tsx:69]: `master` code for that: `!individualReward?.isZero()`.

**Review by aziolek — APPROVED** [review]

**Pull request merged**